app/calculation/index.js

```diff
@@ -2,15 +2,56 @@ const getPaymentBands = require('./get-payment-band')
 const calculateBand = require('./calculate-band')
 const calculateOverall = require('./calculate-overall')
 const { convertStringToPence } = require('./convert-currency')
+const schemeYears = require('./scheme-years')
+
+function transformResults (bandResults) {
+  return Object.values(bandResults.flat().reduce((r, { band, result }) => {
+    if (!r[band]) {
+      r[band] = {
+        band: band,
+        result: []
+      }
+    }
+    result.map((bandResult) => {
+      r[band].result.push(bandResult)
+      return bandResult
+    })
+    return r
+  }, {}))
+}
```

Comment:
This function will like take some time to figure out what it is is doing.  Is there a way to break it down, rename variables or even add comments to make it clearer why it's needed?